### ISSUE TITLE
Adding staging

### ISSUE THREAD

**Issue by christian-oreilly** [lina-usc/pylossless#195]
In the discussion of issue #73, we went a bit on a tangent about staging. I'd like to keep #73 on implementing Artifact Blocking only and move the discussion about adding staging here. My proposal, from #73, is to add a callback like this

```python
def my_staging_function(...):
    ....

staging_kwargs = {...}


ll.set_staging_function(my_staging_functions, staging_kwargs)
```

Setting this to milestone v0.4 because I don't currently have a use case for staging. When addressing this issue, I would like us to have a use case that we can use for the documentation (an example to demonstrate). I'll push the can on this one until we do.

**Comment by scott-huberty** [comment]
Agreed.. Currently I haven't needed a staging function yet because if I do need to do some custom processing beforehand... I  just do it on my `raw` object, before creating a `LosslessPipeline` instance and doing `run_with_raw`(!)
